simbad/command_line/simbad_database.py

```diff
@@ -163,6 +189,151 @@ def create_lattice_db(database):
     np.savez_compressed(database, niggli_data)
 
 
+def create_contaminant_db(database, nproc=2, submit_qtype=None, submit_queue=False):
+    """Create a contaminant database
+
+    Parameters
+    ----------
+    database : str
+       The path to the database folder
+    nproc : int, optional
+       The number of processors [default: 2]
+    submit_qtype : str
+       The cluster submission queue type - currently support SGE and LSF
+    submit_queue : str
+       The queue to submit to on the cluster
+
+    Raises
+    ------
+    RuntimeError
+        dimple.contaminants.prepare module not available
+    RuntimeError
+       Windows is currently not supported
+    """
+
+    try:
+        import dimple.contaminants.prepare
+    except ImportError:
+        msg = "This feature will be available with the next dimple update"
+        raise RuntimeError(msg)
+
+    if CUSTOM_PLATFORM == "windows":
+        msg = "Windows is currently not supported"
+        raise RuntimeError(msg)
+
+    dimple.contaminants.prepare.main(verbose=False)
+
+    simbad_dat_path = os.path.join(database, '*', '*', '*', '*.dat')
+    existing_dat_files = [os.path.basename(f).split('.')[0] for f in glob.glob(simbad_dat_path)]
+    erroneous_files = ['4V43']
+    dimple_files = ['cached', 'data.json', 'data.py']
+
+    with open("data.json") as data_file:
+        data = json.load(data_file)
+
+    results = []
+    for child in data["children"]:
+        try:
+            for child_2 in child["children"]:
+                space_group = child_2["name"].replace(" ", "")
+                for child_3 in child_2["children"]:
+                    pdb_code = child_3["name"].split()[0]
+                    if pdb_code in existing_dat_files or pdb_code in erroneous_files:
+                        continue
+                    uniprot_name = child["name"]
+                    uniprot_mnemonic = uniprot_name.split('_')[1]
+                    score = ContaminantSearchResult(pdb_code, space_group, uniprot_name, uniprot_mnemonic)
+                    results.append(score)
+        except KeyError:
+            pass
+
+    if len(results) == 0:
+        logger.info("Contaminant database up to date")
+    else:
+        logger.info(
+            "%d new entries were found in the contaminant database, "
+            + "updating SIMBAD database", len(results)
+        )
+
+        if "MRD_DB" in os.environ:
+            morda_installed_through_ccp4 = True
```

Comment:
Maybe we can make this some SIMBAD constant. I'm sure in the database script we're also checking for this. Maybe place it in `simbad.util`?

Reply:
I think we only check this in `simbad.command_line.simbad_database`. Ronan has asked me to remove the download_morda() option from simbad_database at some point in due course at which point I might refactor this.

Reply:
I'd still make it somewhere a constant. Maybe top of this file?